src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/MaskDesignerDialogTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.ComponentModel.Design;
+using Moq;
+
+namespace System.Windows.Forms.Design.Tests;
+
+public sealed class MaskDesignerDialogTests : IDisposable
+{
+    private readonly MaskedTextBox _maskedTextBox = new MaskedTextBox();
+    private readonly MaskDesignerDialog _dialog;
+
+    public MaskDesignerDialogTests()
+    {
+        _dialog = new MaskDesignerDialog(_maskedTextBox, null);
+    }
+
+    public void Dispose()
+    {
+        _maskedTextBox.Dispose();
+        _dialog.Dispose();
+    }
+
+    [Fact]
+    public void Constructor_ValidMaskedTextBox_UsesProvidedMaskedTextBox()
+    {
+        TextBox txtBoxMask = _dialog.TestAccessor().Dynamic._txtBoxMask;
+
+        _dialog.Mask.Should().Be(_maskedTextBox.Mask);
+        txtBoxMask.Text.Should().Be(_maskedTextBox.Mask);
+    }
+
+    [Fact]
+    public void ValidatingTypeProperty_ShouldBeSetCorrectly()
+    {
+        _dialog.TestAccessor().Dynamic._maskedTextBox.ValidatingType = typeof(DateTime);
+        _dialog.TestAccessor().Dynamic.btnOK_Click(null, EventArgs.Empty);
+
+        _dialog.ValidatingType.Should().Be(typeof(DateTime));
+    }
+
+    [Fact]
+    public void MaskDescriptorsEnumerator_ShouldReturnCorrectDescriptors()
+    {
+        Collections.IEnumerator enumerator = _dialog.MaskDescriptors;
+
+        List<MaskDescriptor> descriptors = new List<MaskDescriptor>();
```

Comment:
Please use target type new
```suggestion
        List<MaskDescriptor> descriptors = new();
```